src/patient/PatientInfoFields.tsx

```diff
@@ -0,0 +1,84 @@
+import { Field, ErrorMessage, FormikState } from 'formik';
+import { differenceInYears } from 'date-fns';
+
+export interface PatientInfo {
+  name: string;
+  dateOfBirth: string;
+  eye: 'left' | 'right';
+}
+
+const calcAge = (dateOfBirth: string | undefined, error: string | undefined) =>
+  dateOfBirth && !error
+    ? differenceInYears(new Date(), new Date(dateOfBirth)) || ''
+    : '';
+
+export const PatientInfoFields = ({
+  errors,
+  values,
+  touched
+}: FormikState<PatientInfo | undefined>) => (
```

Comment:
The goal here if for this "Patient Info" component to only know about its own props. Does this make sense? Is this the correct way to do it?

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys